src/contracts/core/AVSDirectory.sol

```diff
@@ -132,31 +351,157 @@ contract AVSDirectory is
      *
      */
 
+    /// @inheritdoc IAVSDirectory
+    function operatorSetsMemberOfAtIndex(address operator, uint256 index) external view returns (OperatorSet memory) {
+        return _decodeOperatorSet(_operatorSetsMemberOf[operator].at(index));
+    }
+
+    /// @inheritdoc IAVSDirectory
+    function operatorSetMemberAtIndex(OperatorSet memory operatorSet, uint256 index) external view returns (address) {
+        return _operatorSetMembers[_encodeOperatorSet(operatorSet)].at(index);
+    }
+
+    /// @inheritdoc IAVSDirectory
+    function getNumOperatorSetsOfOperator(
+        address operator
+    ) external view returns (uint256) {
+        return _operatorSetsMemberOf[operator].length();
+    }
+
+    /// @inheritdoc IAVSDirectory
+    function getOperatorSetsOfOperator(
+        address operator,
+        uint256 start,
+        uint256 length
+    ) public view returns (OperatorSet[] memory operatorSets) {
+        uint256 maxLength = _operatorSetsMemberOf[operator].length() - start;
+        if (length > maxLength) length = maxLength;
+        operatorSets = new OperatorSet[](length);
+        for (uint256 i; i < length; ++i) {
+            operatorSets[i] = _decodeOperatorSet(_operatorSetsMemberOf[operator].at(start + i));
+        }
+    }
+
+    /// @inheritdoc IAVSDirectory
+    function getOperatorsInOperatorSet(
+        OperatorSet memory operatorSet,
+        uint256 start,
+        uint256 length
+    ) external view returns (address[] memory operators) {
+        bytes32 encodedOperatorSet = _encodeOperatorSet(operatorSet);
+        uint256 maxLength = _operatorSetMembers[encodedOperatorSet].length() - start;
+        if (length > maxLength) length = maxLength;
+        operators = new address[](length);
+        for (uint256 i; i < length; ++i) {
+            operators[i] = _operatorSetMembers[encodedOperatorSet].at(start + i);
+        }
+    }
+
+    /// @inheritdoc IAVSDirectory
+    function getStrategiesInOperatorSet(
+        OperatorSet memory operatorSet
+    ) external view returns (IStrategy[] memory strategies) {
+        bytes32 encodedOperatorSet = _encodeOperatorSet(operatorSet);
+        uint256 length = _operatorSetStrategies[encodedOperatorSet].length();
+
+        strategies = new IStrategy[](length);
```

Comment:
Can we just do 
```
address[] memory strategyAddresses = _operatorSetStrategies[encodedOperatorSet].values();
        
assembly {
      strategies := strategyAddresses
}
```

Reply:
oh, good call! did not see that method in the library